projects/wildcredit/index.js

```diff
@@ -0,0 +1,114 @@
+const sdk = require("@defillama/sdk");
+const erc20 = require("../helper/abis/erc20.json");
+const Web3 = require("web3");
+const abi = require("./abi.json");
+const json = require("./abis.json");
+
+require("dotenv").config()
+
+let provider = new Web3.providers.WebsocketProvider(
+  process.env.INFURA_WWS
+);
+
+let web3 = new Web3(provider);
+
+const pair_factory = "0x23B74796b72f995E14a5e3FF2156Dad9653256Cf";
+
+const calcTvl = async (balances, block, token, balance) => {
+
+  const factory = new web3.eth.Contract(json, pair_factory);
+  const START_BLOCK = 12867493;
+  const END_BLOCK = await web3.eth.getBlockNumber();
+  const events = await factory.getPastEvents("PairCreated", {
+    fromBlock: START_BLOCK,
+    toBlock: END_BLOCK,
+  });
+  const pairs = events.map(el => el.returnValues.pair);
+
+  const tokens = (await sdk.api.abi.multiCall({
+    abi: token,
+    calls: pairs.map(pair => ({
+      target: pair
+    })),
+    block
+  })).output.map(t => t.output);
+
+  if (balance == erc20.balanceOf) {
+    const balancePairs = (await sdk.api.abi.multiCall({
+      abi: balance,
+      calls: pairs.map((pair, idx) => ({
+        target: tokens[idx],
+        params: pair
+      })),
+      block
+    })).output.map(bp => bp.output);
+
+    for (let index = 0; index < pairs.length; index++) {
+      sdk.util.sumSingleBalance(
+        balances,
+        tokens[index],
+        balancePairs[index]
+      );
+    }
+  } else {
+    const totalDebt = (await sdk.api.abi.multiCall({
```

Comment:
Not sure how this works but does this mean that we are including borrowed money into our TVL?

Reply:
Yes, we have to include it, isn't? I think the borrowed portion should be as part of tvl because it is counted as protocol funds.